### DISCUSSION TITLE
Masking example

### DISCUSSION THREAD

**Discussion by leastbad** [NVlabs/instant-ngp#407] [Q&A] [answered]
In the project README, you suggest that it's possible to provide mask PNGs. I find the language just abstract enough that it's not 100% clear exactly what it actually requires.

When I think of a mask, I think of this:

<img width="561" alt="chrome_wUPTjxj1Ds" src="https://user-images.githubusercontent.com/38150464/160559478-85749c25-25e0-4150-8235-d1dfed74091d.png">

Is that what you mean?

**Comment by Tom94** [comment, marked as answer]
The colors would have to be inverted for this mask to work with our pipeline. Each pixel can be either masked away (non-zero valued) or content (zero valued, i.e. 0x00000000). Cheers!

**Reply by leastbad** [reply]
Interesting. Let me ask a similar question from a different angle: is the intention of these masks to show what a capture space looks like without a subject in frame, or is it to isolate a subject from whatever background is present?

Both are very useful in different situations.

**Reply by Tom94** [reply]
You can use it for both -- in the case of subject removal, your initial mask is perfect without inversion, actually.

That said, background removal can also be done by using the alpha channel of the training data. This will explicitly supervise transparency in the background, rather than ignoring background pixels, which is what happens in this codebase when using a mask.
